### PR TITLE
Summaries subdialog

### LINKED ISSUE #4497: Summaries - first and last and nth
The summaries dialogue (and hence climatic summaries has changed a lot. It is looking very impressive now. 

We now have First and Last and nth. What remains is the add the **OrderBy** facility to these summaries. I suggest we keep First/Last/nth where they are now. 

In addition we repeat them in the 2-variable tab - with the OrderBy label for that block of summaries. So it is effectively another statistic - I hope that is OK?

### PR DESCRIPTION
**Pull request by Muthenya**
I have taken up the issue from @shadrackkibet. This fixes  #4497.

### CODE REVIEW

File: instat/sdgSummaries.vb [hunk 1]
```diff
@@ -344,6 +347,15 @@ Public Class sdgSummaries
         End If
     End Sub
 
+    Private Sub OrderByCheckEnabled()
+        If ucrChkFirst.Checked OrElse ucrChkLast.Checked OrElse ucrChknth.Checked Then
+            ucrChkOrderBy.Enabled = True
+        Else
+            ucrChkOrderBy.Enabled = False
+        End If
```
Comment (shadrackkibet) [inline instat/sdgSummaries.vb]: let's do this automatically through linking of controls.

**Comment by dannyparsons** [comment]
Seems to work well. Two things:

- When I check order by and then uncheck all the summaries, order by is disabled but still checked and the selector receiver still visible. I think it should be unchecked with controls hidden
- The `n` label and text box doesn't seem to line up with the `nth` label

File: instat/sdgSummaries.vb (hunk 1 above)
Comment (Muthenya) [reply instat/sdgSummaries.vb]: @shadrackkibet how should I do this and still ensure that the Check Order by checkbox is disabled and unchecked when all the summaries are unchecked?

Comment (dannyparsons) [reply instat/sdgSummaries.vb]: I think that needs to be done manually.

Comment (shadrackkibet) [reply instat/sdgSummaries.vb]: okay

File: instat/sdgSummaries.resx [hunk 2]
```diff
@@ -1663,7 +1663,7 @@
     <value>3</value>
   </data>
   <data name="ucrInputN.Location" type="System.Drawing.Point, System.Drawing">
-    <value>279, 17</value>
+    <value>280, 19</value>
```
Comment (maxwellfundi) [inline instat/sdgSummaries.resx]: @Muthenya  this is a minor one but you need to move this a little more 1px to be aligned properly.

**Comment by maxwellfundi** [comment]
@dannyparsons  you can have a look at this now.

File: instat/sdgSummaries.resx (hunk 2 above)
Comment (dannyparsons) [reply instat/sdgSummaries.resx]: I agree

**Comment by dannyparsons** [comment]
Working well now, great.

These three controls still seem completley out of line though:

image

**Comment by Muthenya** [comment]
I'm on it.

**Comment by Muthenya** [comment]
I guess this is now okay.

image

**Pull request merged**